tests/ignite/metrics/test_psnr.py

```diff
@@ -0,0 +1,288 @@
+import os
+
+import numpy as np
+import pytest
+import torch
+from skimage.metrics import peak_signal_noise_ratio as ski_psnr
+
+import ignite.distributed as idist
+from ignite.exceptions import NotComputableError
+from ignite.metrics import PSNR
+from ignite.utils import manual_seed
+
+
+def test_zero_div():
+    psnr = PSNR()
+    with pytest.raises(NotComputableError, match="PSNR must have at least one example before it can be computed"):
+        psnr.compute()
+
+
+def test_invalid_psnr():
+    y_pred = torch.rand(1, 3, 8, 8)
+    y = torch.rand(1, 3, 8, 8)
+
+    with pytest.raises(TypeError, match=r"Expected y_pred and y to have the same data type."):
+        psnr = PSNR()
+        psnr.update((y_pred, y.double()))
+
+    with pytest.raises(ValueError, match=r"Expected y_pred and y to have the same shape."):
+        psnr = PSNR()
+        psnr.update((y_pred, y.squeeze(dim=0)))
+
+    with pytest.raises(ValueError, match=r"y has intensity values outside the range expected"):
+        psnr = PSNR()
+        psnr.update((y_pred, y))
+        # to catch ValueError for this batch
+        psnr.update((y_pred, y + 1.0))
+
+
+def _test_psnr_per_image(y_pred, y, data_range, device):
+    psnr = PSNR(data_range=data_range, device=device)
+    psnr.update((y_pred, y))
+
+    np_y_pred = y_pred.cpu().numpy()
+    np_y = y.cpu().numpy()
+    np_psnr = []
+    for np_y_pred_, np_y_ in zip(np_y_pred, np_y):
+        np_psnr.append(ski_psnr(np_y_, np_y_pred_, data_range=data_range))
+
+    assert np.allclose(psnr._sum_of_batchwise_psnr.cpu().numpy(), np_psnr)
```

Comment:
@vfdev-5 do you think this test is still required after we move `torch.sum()` to `update()`?
We are comparing a list of batch results vs a list of batch results.
But after we move `torch.sum()` to `update()`, we are comparing sum values which are being covered by the below tests

Reply:
Yes, let's remove it.